tests/variable/variable.cpp

```diff
@@ -438,6 +438,55 @@ TEST(Variable, getVariableMethods) {
     EXPECT_EQ(nullptr, static_cast<const libcellml::Component>(c).getVariable("doesntexist"));
 }
 
+
+TEST(Variable, takeVariableMethods) {
+    const std::string in = "valid_name";
+    libcellml::Component c;
+    c.setName(in);
+
+    libcellml::VariablePtr v1 = std::make_shared<libcellml::Variable>();
+    v1->setName("variable1");
+    c.addVariable(v1);
+    libcellml::VariablePtr v2 = std::make_shared<libcellml::Variable>();
+    v2->setName("variable2");
+    c.addVariable(v2);
+    libcellml::VariablePtr v3 = std::make_shared<libcellml::Variable>();
+    v3->setName("variable3");
+    c.addVariable(v3);
+    libcellml::VariablePtr v4 = std::make_shared<libcellml::Variable>();
+    v4->setName("variable4");
+    c.addVariable(v4);
+
+    // Take by index
+    libcellml::VariablePtr vMethod2 = c.takeVariable(0);
+    std::string a2 = vMethod2->getName();
+    EXPECT_EQ("variable1", a2);
```

Comment:
Re `vMethod2`: I would rename it to `tv1` (for take variable nb. 1).
Re `a2`: I would rename it to `tvn1` (for take variable name nb. 1).